### ISSUE TITLE
[JIT] Eval / Updating Training doesnt work on traced modules

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#43661]
## 🐛 Bug
```
        class MyModel(torch.nn.Module):
            def __init__(self):
                super(MyModel, self).__init__()
                self.layer1 = torch.nn.Conv2d(3, 32, 1, 1)

            def forward(self, x):
                x = self.layer1(x)
                return x

        model = MyModel().eval()
        shape = [1, 3, 32, 32]
        stride = (3072, 1, 96, 3)
        dummy_input = torch.randn(shape).as_strided(shape, stride)
        traced_module = torch.jit.trace(model, dummy_input)
        traced_module.eval()
        print(traced_module.training) # True
```

cc @gmagogsfm

**Comment by gmagogsfm** [comment]
The issue is that `traced_module.training` should print false instead of true.

**Comment by gmagogsfm** [comment]
Potential change: add a warning when trace module attr is changed due to mode change

**Comment by skariel** [comment]
This is a problem when freezing models, it requires eval mode (i.e. training == False)

**Comment by eellison** [comment]
Fixed with https://github.com/pytorch/pytorch/pull/47211
